### PR TITLE
Add missing German translation

### PR DESCRIPTION
**Pull request by Siedlerchr**
Refs #3174 

- [ ] Change in CHANGELOG.md described
- [ ] Tests created for changes
- [ ] Screenshots added (for bigger UI changes)
- [ ] Manually tested changed features in running JabRef
- [ ] Check documentation status (Issue created for outdated help page at [help.jabref.org](https://github.com/JabRef/help.jabref.org/issues)?)
- [ ] If you changed the localization: Did you run `gradle localizationUpdate`?

### CODE REVIEW

**Comment by mlep** [comment]
@Siedlerchr: There are a few space characters among half a dozen of localized strings (not this commit I believe). Please, could you take this commit as an opportunity to fix them?

**Comment by Siedlerchr** [comment]
@mlep Ah, thanks for the hint! I searched and replaced all occurences of them

**Comment by mlep** [comment]
@Siedlerchr Great. Thank you!

File: src/main/resources/l10n/JabRef_de.properties [hunk 1]
```diff
@@ -2225,11 +2225,11 @@ Opens_JabRef's_website=Öffnet_JabRefs_Webseite
...
-Finished_downloading_full_text_document_for_entry_%0.=Download des Volltextdokuments für Eintrag %0 komplett
-Full_text_document_download_failed_for_entry_%0.=Download des Volltextdokuments für Eintrag %0 fehlgeschlagen
-Look_up_full_text_documents=Volltextdokumente suchen
-You_are_about_to_look_up_full_text_documents_for_%0_entries.=Sie werden gleich Volltextdokumente für %0 Einträge suchen
-last_four_nonpunctuation_characters_should_be_numerals=Die letzten vier Nichtinterpunktionszeichen sollten Numerale sein
+Finished_downloading_full_text_document_for_entry_%0.=Download_des_Volltextdokuments_für_Eintrag_%0_komplett
+Full_text_document_download_failed_for_entry_%0.=Download_des_Volltextdokuments_für_Eintrag_%0_fehlgeschlagen
+Look_up_full_text_documents=Volltextdokumente_suchen
+You_are_about_to_look_up_full_text_documents_for_%0_entries.=Sie_werden_gleich_Volltextdokumente_für_%0_Einträge_suchen
+last_four_nonpunctuation_characters_should_be_numerals=Die_letzten_vier_Nichtinterpunktionszeichen_sollten_Numerale_sein
```
Comment (lynyus) [inline src/main/resources/l10n/JabRef_de.properties]: Statt Numerale -> Ziffern?

File: src/main/resources/l10n/JabRef_de.properties [hunk 2]
```diff
@@ -2323,20 +2323,20 @@ Locked=Gesperrt
...
+This_file_was_found_automatically._Do_you_want_to_link_it_to_this_entry?=Diese_Datei_wurde_automatisch_gefunden._Möchten_Sie_sie_dem_Eintrag_zuordnen?
+Names_are_not_in_the_standard_BibTeX_format.=Namen_entsprechen_nicht_dem_Standard_BibTeX_Format
+
+Delete_the_selected_file_permanently_from_disk,_or_just_remove_the_file_from_the_entry?_Pressing_Delete_will_delete_the_file_permanently_from_disk.=Die_ausgewählte_Datei_unwiderruflich_löschen_oder_nur_vom_Eintrag_entfernen?_Drücken_sie_Entfernen,_um_die_Datei_von_der_Festplatte_zu_löschen.
+Delete_'%0'=Lösche_Datei_'%0'
+Delete_from_disk=Auf_der_Festplatte_löschen
```
Comment (lynyus) [inline src/main/resources/l10n/JabRef_de.properties]: **Von** der Festplatte?

File: src/main/resources/l10n/JabRef_de.properties [hunk 3]
```diff
@@ -2323,20 +2323,20 @@ Locked=Gesperrt
...
-The_group_name_contains_the_keyword_separator_"%0"_and_thus_probably_does_not_work_as_expected.=
-There_exists_already_a_group_with_the_same_name.=
+To_improve_the_user_experience,_we_would_like_to_collect_anonymous_statistics_on_the_features_you_use._We_will_only_record_what_features_you_access_and_how_often_you_do_it._We_will_neither_collect_any_personal_data_nor_the_content_of_bibliographic_items._If_you_choose_to_allow_data_collection,_you_can_later_disable_it_via_Options_->_Preferences_->_General.=Zur_Verbesserung_der_Benutzererfahrung_würden_wir_gerne_anonyme_Statistiken_über_die_Features_sammeln,_die_Sie_benutzen._Wir_zeichnen_nur_auf,_welche_Features_Sie_nutzen_und_wie_oft_Sie_diese_benutzen._Es_werden_keinerlei_persönliche_Informationen_über_Sie_oder_den_Inhalt_Ihrer_Bibliografieeinträge_gesammelt._Die_einmal_erlaubte_Datenaufzeichnnung_kann_jederzeit_unter_Optionen->Einstellungen->Allgemein_deaktiviert_werden.
+This_file_was_found_automatically._Do_you_want_to_link_it_to_this_entry?=Diese_Datei_wurde_automatisch_gefunden._Möchten_Sie_sie_dem_Eintrag_zuordnen?
+Names_are_not_in_the_standard_BibTeX_format.=Namen_entsprechen_nicht_dem_Standard_BibTeX_Format
```
Comment (lynyus) [inline src/main/resources/l10n/JabRef_de.properties]: BibTeX-Format?

File: src/main/resources/l10n/JabRef_de.properties (hunk 2 above)
Comment (Siedlerchr) [reply src/main/resources/l10n/JabRef_de.properties]: I found "von" and "auf" as both possible when searching for it...

**Pull request merged**